pennylane/transforms/zx/converter.py

```diff
@@ -0,0 +1,577 @@
+        function.
+
+        Copyright (C) 2018 - Aleks Kissinger and John van de Wetering
+    """
+    # If it is a simple operation just transform it to a tape
+    return _to_zx(QuantumScript([qscript]))
+
+
+@to_zx.tape_transform
+def _to_zx(qscript, expand_measurements=False):
+    """Private function to convert a PennyLane tape to a `PyZX graph <https://pyzx.readthedocs.io/en/latest/>`_ ."""
+    # Avoid to make PyZX a requirement for PennyLane.
+    try:
+        # pylint: disable=import-outside-toplevel
+        import pyzx
+        from pyzx.circuit.gates import TargetMapper
+        from pyzx.graph import Graph
+
+    except ImportError as Error:
+        raise ImportError(
+            "This feature requires PyZX. It can be installed with: pip install pyzx"
+        ) from Error
+
+    # Dictionary of gates (PennyLane to PyZX circuit)
+    gate_types = {
+        "PauliX": pyzx.circuit.gates.NOT,
+        "PauliZ": pyzx.circuit.gates.Z,
+        "S": pyzx.circuit.gates.S,
+        "T": pyzx.circuit.gates.T,
+        "Hadamard": pyzx.circuit.gates.HAD,
+        "RX": pyzx.circuit.gates.XPhase,
+        "RZ": pyzx.circuit.gates.ZPhase,
+        "PhaseShift": pyzx.circuit.gates.ZPhase,
+        "SWAP": pyzx.circuit.gates.SWAP,
+        "CNOT": pyzx.circuit.gates.CNOT,
+        "CZ": pyzx.circuit.gates.CZ,
+        "CRZ": pyzx.circuit.gates.CRZ,
+        "CH": pyzx.circuit.gates.CHAD,
+        "CCZ": pyzx.circuit.gates.CCZ,
+        "Toffoli": pyzx.circuit.gates.Tofolli,
+    }
+    # Create the graph, a qubit mapper, the classical mapper stays empty as PennyLane does not support classical bits.
+    graph = Graph(None)
+    q_mapper = TargetMapper()
+    c_mapper = TargetMapper()
+
+    # Map the wires to consecutive wires
+
+    consecutive_wires = Wires(range(len(qscript.wires)))
+    consecutive_wires_map = OrderedDict(zip(qscript.wires, consecutive_wires))
+    qscript = qml.map_wires(input=qscript, wire_map=consecutive_wires_map)
+
+    inputs = []
+
+    # Create the qubits in the graph and the qubit mapper
+    for i in range(len(qscript.wires)):
+        vertex = graph.add_vertex(VertexType.BOUNDARY, i, 0)
+        inputs.append(vertex)
+        q_mapper.set_prev_vertex(i, vertex)
+        q_mapper.set_next_row(i, 1)
+        q_mapper.set_qubit(i, i)
+
+    # Expand the qscript to be compatible with PyZX and add rotations first for measurements
+    stop_crit = qml.BooleanFn(lambda obj: obj.name in gate_types)
+    qscript = qml.tape.tape.expand_tape(
+        qscript, depth=10, stop_at=stop_crit, expand_measurements=expand_measurements
+    )
+
+    expanded_operations = []
+
+    # Define specific decompositions
+    for op in qscript.operations:
+        if op.name == "RY":
+            theta = op.data[0]
+            decomp = [
+                qml.RX(np.pi / 2, wires=0),
+                qml.RZ(theta + np.pi, wires=0),
+                qml.RX(np.pi / 2, wires=0),
+                qml.RZ(3 * np.pi, wires=0),
```

Comment:
is the 3pi intentional here?

Reply:
I used it in the test just be sure that something outside [0, 2pi] would work.